cmd/hdfs/ls.go

```diff
@@ -100,30 +129,67 @@ func printDir(client *hdfs.Client, dir string, long, all, humanReadable bool) {
 		}
 	}
 
-	var partial []os.FileInfo
+	var partial, files []os.FileInfo
+	var fileCount = 0
 	for ; err != io.EOF; partial, err = dirReader.Readdir(100) {
 		if err != nil {
 			fatal(err)
 		}
+		fileCount += len(partial)
+		files = append(files, partial...)
+	}
 
-		printFiles(tw, partial, long, all, humanReadable)
+	if recursive {
+		if long {
+			if recursionLevel > 0 {
+				fmt.Fprintln(tw)
+			}
+			fmt.Fprintf(tw, "%s:\n", dir)
+			fmt.Fprintln(tw, "total", fileCount)
+		} else {
+			if recursionLevel > 0 {
+				fmt.Println()
+			}
+			fmt.Printf("%s:\n", dir)
+			fmt.Println("total", fileCount)
+		}
+		recursionLevel++
 	}
 
+	printFiles(client, tw, dir, files, long, all, recursive, humanReadable, selfOnly)
```

Comment:
Instead of recursing here, let's move that functionality higher up and use `Walk`. You can follow the example for `chmod`: https://github.com/colinmarc/hdfs/blob/master/cmd/hdfs/chmod.go#L35-L51